Actively/Actively/ActivelyApp/Services/EntityService/GameService.cs

```diff
@@ -0,0 +1,117 @@
+﻿using ActivelyApp.CustomExceptions;
+using ActivelyApp.Models.EntityDto;
+using ActivelyDomain.Entities;
+using ActivelyInfrastructure.Repositories.EntityRepositories.GameRepository;
+using AutoMapper;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Resources;
+
+namespace ActivelyApp.Services.EntityService
+{
+
+    public class GameService : IGameService
+    {
+        private readonly IGameRepository _gameRepository;
+        private readonly IMapper _mapper;
+
+        public GameService(IGameRepository gameRepository, IMapper mapper)
+        {
+            _gameRepository = gameRepository;
+            _mapper = mapper;
+        }
+        public async Task<IEnumerable<GameDto>> GetAll()
+        {
+            IEnumerable<Game> games = null;
+            try
+            {
+                games = await _gameRepository.GetAll() ?? new List<Game>();
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+            var gamesDto = _mapper.Map<IEnumerable<GameDto>>(games);
+            return gamesDto;           
+        }
+
+        public async Task<GameDto?> GetById(int id)
+        {
+            Game game = null;
+            try
+            {
+                game = await _gameRepository.GetById(id);
+            }
+            catch (Exception)
+            {
+                //log
+            }
+            if (game == null)
+                return null;
+            var gameDto = _mapper.Map<GameDto>(game);
+            return gameDto;
+        }
+
+        public async Task Delete(int id)
+        {
+            Game gameToDelete = null;
+            try
+            {
+                gameToDelete = await _gameRepository.GetById(id);
+                if (gameToDelete == null)
+                {
+                    throw new NotFoundEntityException(Common.GameNotExistsError);
+                }
+                await _gameRepository.Delete(gameToDelete);
+                await _gameRepository.Save();
+            }
+            catch (NotFoundEntityException)
+            {
+                //log
+            }
```

Comment:
Ogolnie ladnie to wyglada, ale widze jedna rzecz, napisze po polsku:
zrobie to na przykladzie tego Taska Delete(int id).

uzywamy tego Taska GmaeService.Delete() np w GameController kod poniżej:

 [HttpDelete]
        public async Task<IActionResult> Delete(int id)
        {
            try
            {
                await _gameService.Delete(id);
            }
            catch(**NotFoundEntityException**)
            {
                return StatusCode(StatusCodes.Status404NotFound, new Response
                { Type = ResponseType.Error, Status = Common.Error, Message = Common.GameNotExistsError, IsSuccess = false });
            }


w kontrolerze chcesz złapac wyjątek **NotFoundEntityException**, który zostanie zjedzony niepotrzebnie w GameServisie. Trzeba to pozmieniac bo teraz nigdy nie dojdziesz do catch'a w kontrolerze.
W efekcie dostalem
![image](https://github.com/Woselko/Actively/assets/76818798/6358b51e-3b7f-442a-84ab-35c16fe74487)
z pustą bazą danych :P

Reply:
i generalnie we wszystkich tych serwisach zle handlujesz wyjatki, zjadamy je i w efekcie return succes jest

Reply:
tak samo w testach
                    _mockGameService.Setup(service => service.Delete(id)).ThrowsAsync(new NotFoundEntityException(Common.GameNotExistsError));
                    
realnie mock czy tez serwis nigdy nie rzucilby tym wyjatkiem, bo zostal zjedzony, chyab ze bylby rethrow